site-manage/site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/impl/SiteManageGroupSectionRoleHandler.java

```diff
@@ -854,268 +870,376 @@ public String processCancelDelete()
     }
     
     /**
-     * atuo create group(s) based on selected roster(s) or role(s)
+     * SAK-29373 - Common validation algorithm for both roster and role based random groups
+     * 
+     * @param option
+     * @param isGroupSplit
+     * @param numToSplitForGroups
+     * @param numToSplitForUsers
+     * @param groupTitleForGroups
+     * @param groupTitleForUsers
+     * @return 0 if not for random groups, -1 if for random groups and validation fails. Otherwise, returns the 
+     * integer value of the number of groups or number of users per group requested.
+     */
+    private int validateAutoGroupsFields( int option, boolean isGroupSplit, String numToSplitForGroups, String numToSplitForUsers,
+                                          String groupTitleForGroups, String groupTitleForUsers)
+    {
+        int intToSplit = 0;
+        if( OPTION_ASSIGN_RANDOM == option )
+        {
+            String numToSplit = isGroupSplit ? numToSplitForGroups : numToSplitForUsers;
+            String groupTitle = isGroupSplit ? groupTitleForGroups : groupTitleForUsers;
+            if( numToSplit == null )
+            {
+                if( isGroupSplit )
+                {
+                    messages.addMessage( new TargettedMessage( "numToSplit.group.empty.alert", "numToSplit" ) );
+                }
+                else
+                {
+                    messages.addMessage( new TargettedMessage( "numToSplit.user.empty.alert", "numToSplit" ) );
+                }
+                return -1;
+            }
+            else
+            {
+                try
+                {
+                    intToSplit = Integer.parseInt( numToSplit );
+                    if( intToSplit <= 0 )
+                    {
+                        if( isGroupSplit )
```

Comment:
Would be cleaner if this section (and the one repeated below for NFE) was extracted into a method.